tests/integration/nm/vlan_test.py

```diff
@@ -61,11 +59,15 @@ def _vlan_interface(ctx, state):
         _delete_vlan(ctx, _get_vlan_ifname(state))
 
 
-def _get_vlan_current_state(nm_plugin, vlan_desired_state):
-    nm_plugin.refresh_content()
+def _get_vlan_current_state(vlan_desired_state):
     ifname = _get_vlan_ifname(vlan_desired_state)
-    nmdev = nm_plugin.context.get_nm_dev(ifname)
-    return nm.vlan.get_info(nmdev) if nmdev else {}
+    state = statelib.show_only((ifname,))
+    vlan_state = {}
+    if state[Interface.KEY]:
+        iface_state = state[Interface.KEY][0]
+        vlan_state = {iface_state[VLAN.CONFIG_SUBTREE]}
```

Comment:
The key is missing here: `vlan_state = {VLAN.CONFIG_SUBTREE: iface_state[VLAN.CONFIG_SUBTREE]}`. My bad, I missed it in my suggestion too.